src/templates/finops-hub/modules/dataFactory.bicep

```diff
@@ -314,6 +320,458 @@ resource trigger_exportContainer 'Microsoft.DataFactory/factories/triggers@2018-
   }
 }
 
+resource trigger_configContainer 'Microsoft.DataFactory/factories/triggers@2018-06-01' = {
+  name: safeConfigContainerName
+  parent: dataFactory
+  dependsOn: [
+    stopHubTriggers
+  ]
+  properties: {
+    annotations: []
+    pipelines: [
+      {
+        pipelineReference: {
+          referenceName: pipeline_setup.name
+          type: 'PipelineReference'
+        }
+        parameters: {
+          FolderName: '@triggerBody().folderPath'
+          FileName: '@triggerBody().fileName'
+        }
+      }
+    ]
+    type: 'BlobEventsTrigger'
+    typeProperties: {
+      blobPathBeginsWith: '/${configContainerName}/blobs/'
+      blobPathEndsWith: 'settings.json'
+      ignoreEmptyBlobs: true
+      scope: storageAccount.id
+      events: [
+        'Microsoft.Storage.BlobCreated'
+      ]
+    }
+  }
+}
+
+//------------------------------------------------------------------------------
+// Pipelines
+//------------------------------------------------------------------------------
+
+//------------------------------------------------------------------------------
+// Azure Cost Management add export pipeline
+// Triggered when settings.json is updated.
+// Creates an export in Azure Cost Management.
+//------------------------------------------------------------------------------
+resource pipeline_setup 'Microsoft.DataFactory/factories/pipelines@2018-06-01' = {
+  name: '${safeExportContainerName}_setup'
```

Comment:
Thinking ahead to when we support other clouds, should this be `config_setup` and not tied to `msexports` explicitly?

Reply:
I was thinking we could create othercloud_setup pipeline and subscribe it to the same trigger.
The msexports_setup pipeline will ignore export scopes which don't have the "cloud" attribute set to AzureCloud or AzureUSGovernment.